jme3-alloc/src/main/java/com/jme3/alloc/util/NativeBufferUtils.java

```diff
@@ -24,77 +55,69 @@ private NativeBufferUtils() {
     }
 
     /**
-     * Allocates a buffer of size bytes.
+     * Allocates a buffer of [size] bytes using [void* malloc(size_t)] provided by GNU stdlib.
+     * Note: Maximum size is 32-bit for 32-bit architectures and 64-bit for 64-bit architectures.
      * 
-     * @param size the size of the newly created buffer in bytes (max = 32-bit or 4 bytes)
-     * @return a new java-nio buffer object
+     * @param size the size of the newly created buffer in bytes 
+     * @return a newly created un-initialized {@link java.nio.ByteBuffer} object
+     * @see NativeBufferUtils#clearAlloc(long)
      */
-    public static native Buffer memoryAlloc(final long size);
+    public static native ByteBuffer memoryAlloc(final long size);
 
     /**
-     * Copies the value [value] into each of the first [size] bytes of the object beginning at the 
-     * memory address of [buffer].
+     * Allocates a buffer block of [size] bytes using [void* calloc(size_t)] provided by GNU stdlib, and initializes its elements to zero.
+     * Note: Maximum size is 32-bit for 32-bit architectures and 64-bit for 64-bit architectures.
```

Comment:
Hmm, what is this max size limit? Can't we create a byte buffer of a size more than 8 bytes?

Reply:
or do you mean the max size can be 2^32 (for 32-bit CPU) and 2^64 (for 64-bit CPU)?

Reply:
I cannot recall what is `2` in `2^64`

I was documenting the ability to allocate the size based on the parameter provided, which is long, long is 64-bit number but it's a 32-bit number on 32-bit binaries.....

But reading this again, i am wrong, the parameter is in `bytes` units, so it should be 32 or 64 bytes (and not bits).

Reply:
>I cannot recall what is 2 in 2^64

Long.MAX_VALUE which actually is (2^63) - 1

So it should be:

>Note: Maximum buffer size can be Integer.MAX_VALUE bytes for 32-bit architectures and Long.MAX_VALUE bytes for 64-bit architectures.

Reply:
By the afaik, there is no such limitation for long on java for 32-bit systems. Is this limitation on C code? 
By the way, you may take a look at: https://stackoverflow.com/questions/50996253/in-c-what-is-the-size-of-a-long-for-a-32-bit-machine-and-for-a-64-bit-machine

Maybe you should remove that note as the limit is not platform specific but depends on C implementation as mentioned in above link.

Reply:
And by the way, JME uses an `int` for size:

https://github.com/jMonkeyEngine/jmonkeyengine/blob/4db02fae9dd8ebb213b3a06316a53019ef9b620f/jme3-core/src/main/java/com/jme3/util/BufferAllocator.java#L23

Reply:
> Is this limitation on C code?

In C, `size_t` is defined as `unsigned long int` which is architecture dependent and it's equivalent to `long` (the gnu docs), 32-bit (or 4 bytes) for 32 architectures and 64-bit (or 8 bytes) for 64 architectures.

In java, `jlong` is defined as `__int64` on windows and `long long` on linux, `long long` is 64-bit despite the system type, while `long` is architecture dependent.

But, there is a note on windows `jni_md.h`:
```
// 'long' is always 32 bit on windows so this matches what jdk expects
```
So, here we don't know the architecture of the system from java, if it's a 32-bit binary the `size_t` max number will be a 32-bit number (or 4 bytes or 0x(FF)(FF)(FF)(FF)), if it was a 64-bit binary, the `size_t` will be a 64-bit number....so we have to always use a 64-bit number anyway.

Reply:
I have not tested passing a 64-bit jlong `Long.MAX_VALUE` to a native `long` parameter on a 32-bit system.....i guess it will lead to overflow and the number will be reset to (-2^32), because the native method accepts native `long` which is architecture dependent, so we should document this properly.....that the native method is architecture dependent, if you have any tips to give on documentation of this part, please let me know.

Reply:
> > Note: Maximum size is 32-bit for 32-bit architectures and 64-bit for 64-bit architectures.
> 
> Probably better to change it to:
> 
> > Note: Maximum buffer size can be 2^32 bytes for 32-bit architectures and 2^64 bytes for 64-bit architectures.
> 
> what do you think?

I think this might be the best description for this docs part, this code runs internally a native code which is architecture specific.

Reply:
@Ali-RS okay, i have refactored this, please review the new documentation.

Reply:
Looks good to me, thanks

Reply:
Great, thanks for your contributions, going to merge this soon.